ethcore/private-tx/src/encryptor.rs

```diff
@@ -187,9 +186,8 @@ impl SecretStoreEncryptor {
 
 	/// Try to unlock account using stored passwords
 	fn unlock_account(&self, account: &Address, accounts: Arc<AccountProvider>) -> bool {
-		let passwords = self.config.passwords.clone();
-		for password in passwords {
-			if let Ok(()) = accounts.unlock_account_temporarily(account.clone(), password) {
+		for password in &self.config.passwords {
+			if let Ok(()) = accounts.unlock_account_temporarily(account.clone(), password.clone()) {
```

Comment:
another security flaw,

also this method is duplicated in 2 different implementations

Reply:
Can we do anything about this?

Reply:
Reworked